src/cache/lru-cache.ts

```diff
@@ -32,16 +63,16 @@ export class LRUCache {
       // This is crucial for maintaining the correct order of elements in the cache,
       // which directly impacts which item is considered the least recently used (LRU) and
       // thus eligible for eviction when the cache reaches its capacity.
-      this.cache.delete(key);
+      this.delete(key);
```

Comment:
Ha - good times. I am remembering this implementation now! If you feel this is too much "magical behavior" and would prefer a more explicit implementation of LRU that is 👍 I think with this we do need to stick to a `Map`